test/integration/controller/job/job_controller_test.go

```diff
@@ -265,6 +269,186 @@ var _ = ginkgo.Describe("Job controller for workloads with no queue set", func()
 	})
 })
 
+var _ = ginkgo.Describe("Job controller when waitForPodsReady enabled", func() {
+	type podsReadyTestSpec struct {
+		jobStatus       batchv1.JobStatus
+		beforeCondition *metav1.Condition
+		wantCondition   *metav1.Condition
+	}
+
+	ginkgo.BeforeEach(func() {
+		fwk = &framework.Framework{
+			ManagerSetup: managerSetup(workloadjob.WithWaitForPodsReady(true)),
+			CRDPath:      crdPath,
+		}
+		ctx, cfg, k8sClient = fwk.Setup()
+	})
+	ginkgo.AfterEach(func() {
+		fwk.Teardown()
+	})
+
+	ginkgo.DescribeTable("Single job at different stages of progress towards completion",
+		func(podsReadyTestSpec podsReadyTestSpec) {
+			ginkgo.By("Create a cluster queue")
+			defaultFlavor := testing.MakeResourceFlavor("default").Label(labelKey, "default").Obj()
+			gomega.Expect(k8sClient.Create(ctx, defaultFlavor)).Should(gomega.Succeed())
+			clusterQueue := testing.MakeClusterQueue("cluster-queue").
+				Resource(testing.MakeResource(corev1.ResourceCPU).
+					Flavor(testing.MakeFlavor(defaultFlavor.Name, "5").Obj()).
+					Obj()).Obj()
+
+			ginkgo.By("Create a job")
+			job := testing.MakeJob(jobName, jobNamespace).Parallelism(2).Obj()
+			jobQueueName := "test-queue"
+			job.Annotations = map[string]string{constants.QueueAnnotation: jobQueueName}
+			gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
+			lookupKey := types.NamespacedName{Name: jobName, Namespace: jobNamespace}
+			createdJob := &batchv1.Job{}
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob)).Should(gomega.Succeed())
+
+			ginkgo.By("Admit the workload created for the job")
+			createdWorkload := &kueue.Workload{}
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, lookupKey, createdWorkload)
+			}, framework.Timeout, framework.Interval).Should(gomega.Succeed())
+			createdWorkload.Spec.Admission = &kueue.Admission{
+				ClusterQueue: kueue.ClusterQueueReference(clusterQueue.Name),
+				PodSetFlavors: []kueue.PodSetFlavors{{
+					Flavors: map[corev1.ResourceName]string{
+						corev1.ResourceCPU: defaultFlavor.Name,
+					},
+				}},
+			}
+			gomega.Expect(k8sClient.Update(ctx, createdWorkload)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdWorkload)).Should(gomega.Succeed())
+
+			if podsReadyTestSpec.beforeCondition != nil {
+				ginkgo.By("Update the workload status")
+				apimeta.SetStatusCondition(&createdWorkload.Status.Conditions, *podsReadyTestSpec.beforeCondition)
+				gomega.Expect(k8sClient.Status().Update(ctx, createdWorkload)).Should(gomega.Succeed())
+				gomega.Expect(k8sClient.Get(ctx, lookupKey, createdWorkload)).Should(gomega.Succeed())
+			}
+
+			ginkgo.By("Update the job status to simulate its progress towards completion")
+			createdJob.Status = podsReadyTestSpec.jobStatus
+			gomega.Expect(k8sClient.Status().Update(ctx, createdJob)).Should(gomega.Succeed())
+
+			ginkgo.By("Verify the PodsReady condition is added")
+			gomega.Eventually(func() error {
```

Comment:
```suggestion
			gomega.Eventually(func() metav1.Condition {
```

then you can use `Should(gomega.BeComparableTo(wantCondition))`. This uses `cmp.Diff` underneath

Reply:
done